libs/eavmlib/src/gpio.erl

```diff
@@ -76,62 +94,215 @@ start() ->
+read(_GPIO, GPIONum) ->
+    digital_read(GPIONum).
 
+%% @depreciated Use {@link gpio:set_pin_mode/2} instead.
 -spec set_direction(GPIO :: gpio(), GPIONum :: pin(), Direction :: direction()) -> ok | error.
-set_direction(GPIO, GPIONum, Direction) ->
-    port:call(GPIO, {set_direction, GPIONum, Direction}).
+set_direction(_GPIO, GPIONum, Direction) ->
+    set_pin_mode(GPIONum, Direction).
 
+%% @depreciated Use {@link gpio:digital_write/2} instead.
 -spec set_level(GPIO :: gpio(), GPIONum :: pin(), Level :: level()) -> ok | error.
-set_level(GPIO, GPIONum, Level) ->
-    port:call(GPIO, {set_level, GPIONum, Level}).
+set_level(_GPIO, GPIONum, Level) ->
+    digital_write(GPIONum, Level).
 
+%%-----------------------------------------------------------------------------
+%% @param   GPIO pid that was returned from gpio:start/0
+%% @param   Pin number of the pin to set the interrupt on
+%% @param   Trigger is the state that will trigger an interrupt
+%% @returns ok
+%% @doc     Set a GPIO interrupt
+%%
+%%          Available triggers are none (which is the same as disabling an
+%%          interrupt), rising, falling, both (rising or falling),low, and
+%%          high. When the interrupt is triggered it will send a tuple:
+%%            {gpio_interrupt, Pin}
+%%          to the process that set the interrupt. `Pin` will be the number
+%%          of the pin that triggered the interrupt.
+%% @end
+%%-----------------------------------------------------------------------------
 -spec set_int(GPIO :: gpio(), GPIONum :: pin(), Trigger :: trigger()) -> ok | error.
 set_int(GPIO, GPIONum, Trigger) ->
     port:call(GPIO, {set_int, GPIONum, Trigger}).
 
+%%-----------------------------------------------------------------------------
+%% @param   GPIO pid that was returned from gpio:start/0
+%% @param   Pin number of the pin to remove the interrupt
+%% @returns ok
+%% @doc     Remove a GPIO interrupt
+%%
+%%          Removes an interrupt from the specified pin.
+%% @end
+%%-----------------------------------------------------------------------------
 -spec remove_int(GPIO :: gpio(), GPIONum :: pin()) -> ok | error.
 remove_int(GPIO, GPIONum) ->
     port:call(GPIO, {remove_int, GPIONum}).
 
+%%-----------------------------------------------------------------------------
+%% @param   Pin number of the pin to remove the interrupt
+%% @param   Direction is input, output, or output_od
+%% @returns ok
+%% @doc     Set the operational mode of a pin
+%%
+%%          Pins can be used for input, output, or output with open drain.
+%% @end
+%%-----------------------------------------------------------------------------
 -spec set_pin_mode(GPIONum :: pin(), Direction :: direction()) -> ok | error.
 set_pin_mode(_GPIONum, _Mode) ->
     throw(nif_error).
 
+%%-----------------------------------------------------------------------------
+%% @param   Pin number of the pin to remove the interrupt
+%% @param   Pull is the internal resistor state
+%% @returns ok
```

Comment:
`| error`

Reply:
I see that `| error` needs to be added throughout. I should note that the errors are thrown in each of these cases, not a simple "return" of an error atom. Does that make a difference in how the edoc should be written?